packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -475,8 +479,8 @@ export function renderWithHooks<Props, SecondArg>(
     // example, in the SuspenseList implementation.
     if (
       current !== null &&
-      (current.flags & PassiveStaticEffect) !==
-        (workInProgress.flags & PassiveStaticEffect)
+      (current.flags & StaticMaskEffect) !==
+        (workInProgress.flags & StaticMaskEffect)
     ) {
```

Comment:
@acdlite mentioned a concern that we might not be resetting static flags correctly. This DEV check has been expanded to also cover the static layout and ref flags.